service-userlayer/src/main/java/org/oskari/map/userlayer/input/GPKGParser.java

```diff
@@ -0,0 +1,66 @@
+package org.oskari.map.userlayer.input;
+
+import java.io.File;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.geotools.data.DataStore;
+import org.geotools.data.DataStoreFinder;
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.data.simple.SimpleFeatureSource;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.oskari.map.userlayer.service.UserLayerException;
+
+import fi.nls.oskari.service.ServiceException;
+
+public class GPKGParser implements FeatureCollectionParser {
+    
+    public static final String SUFFIX = "GPKG";
+
+    @Override
+    public String getSuffix() {
+        return SUFFIX;
+    }
+
+    @Override
+    public SimpleFeatureCollection parse(File file, CoordinateReferenceSystem sourceCRS,
+            CoordinateReferenceSystem targetCRS) throws ServiceException {
+        Map<String, Serializable> params = new HashMap<>();
+        params.put("dbtype", "geopkg");
+        params.put("database", file);
+
+        DataStore store = null;
+        SimpleFeatureCollection collection = null;
+        try {
+            store = DataStoreFinder.getDataStore(params);
+            for (String typeName : store.getTypeNames()) {
+                SimpleFeatureSource source = store.getFeatureSource(typeName);
+                SimpleFeatureType sft = store.getSchema(typeName);
+                CoordinateReferenceSystem crs = sft
+                        .getGeometryDescriptor()
+                        .getCoordinateReferenceSystem();
+                if (crs != null) {
+                    sourceCRS = crs;
+                }
+                collection = FeatureCollectionParsers.read(source, sourceCRS, targetCRS);
+                if (!collection.isEmpty()) {
+                    break;
+                }
+            }
+            return collection;
```

Comment:
Throw an exception with approriate error message if this is null instead of returning null causing a weird NPE in `CreateUserLayerHandler` - probably an `UserLayerException`